### PR TITLE
CVE-2011-1122

### PR DESCRIPTION
**Pull request by Nitepone**
- Set curated to true
- Make simple description
- Answer a couple questions

### CODE REVIEW

File: cves/CVE-2011-1122.yml [hunk 1]
```diff
@@ -32,8 +32,16 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: The WebGL implementation in Google Chrome before 9.0.597.107 allows remote
-  attackers to cause a denial of service (out-of-bounds read) via unspecified vectors.
+description: |
+  CVE-2011-1122 allows attempts to read memory outside expected
+  boundaries on the host operating system. This vulnerability specifically
+  exploits mistakes in how vectors were handled. Specifically, it would
+  use an integer overflow to force an out of bounds memory read from the
+  webkit engine.
```
Comment (Pearse4004) [inline cves/CVE-2011-1122.yml]: Possibly give an example of the vulnerability,  as in what would a user do to trigger the vulnerability.

Comment (Nitepone) [reply cves/CVE-2011-1122.yml]: This was caught by a fuzzer test page, in theory this shouldn't be caused by the user, only by a malicious or broken WebGL application that is loaded.

I have tried to better explain what caused the issue in my newest commit.

File: cves/CVE-2011-1122.yml [hunk 2]
```diff
@@ -49,24 +60,24 @@ reviews:
...
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes:
 unit_tested:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1122.yml]: "Answer" should be an explanation of the values given for code and fix (ideally a brief overview of what kind of tests existed/were added).

File: cves/CVE-2011-1122.yml [hunk 3]
```diff
@@ -131,10 +142,10 @@ interesting_commits:
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1122.yml]: Demonstrate that you completed this section by explaining what happened between the VCCs and the fix.

File: cves/CVE-2011-1122.yml [hunk 4]
```diff
@@ -165,38 +177,38 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies:
```
Comment (CatherineOsadciw) [inline cves/CVE-2011-1122.yml]: Even if you think none of the below lessons apply to this vulnerability, you can add your own.

File: cves/CVE-2011-1122.yml (hunk 2 above)
Comment (Nitepone) [reply cves/CVE-2011-1122.yml]: No unit tests were added to the project. This was caught using updated external conformance testing. Should conformance testing be discussed in this section?

File: cves/CVE-2011-1122.yml [hunk 5]
```diff
@@ -49,24 +60,30 @@ reviews:
...
+  :note: 'Changes how GL version is checked based on bug 429053 (5 years later)'
+- :commit: 9b0c43932354d90f3d5e4882695f442bbbb4708c
+  :note: 'Changes how GL version is checked through the entire file'
 vccs:
-- :commit:
-  :note: 
+- :commit: b112298f642b44d190d201a64cdc60963266551a
+  :note: |
+    'Enables vertex attrib 0, the buffer that caused issue in OpenGl ES 2.0
+     "TEST=a bunch more conformance tests pass"'
```
Comment (Moremothansasuke) [inline cves/CVE-2011-1122.yml]: What does this mean? Like what is "vertex attrib 0", what does it do? This seems confusing and very technical as it is, maybe add a less jargon filled description as well if you think that note is important to keep?

Comment (Nitepone) [reply cves/CVE-2011-1122.yml]: Agreed, this was far too technical. I have now summarized it better. What's important is that the changes were made to pass older OpenGL tests; this implementation caused (possibly not yet available) OpenGL ES 2.0 tests to fail in the future.

**Pull request merged**